src/helper/service.cpp

```diff
@@ -31,6 +33,8 @@ service::service(service_identifier id, std::shared_ptr<engine> engine,
         throw std::runtime_error("invalid engine");
     }
 
+    interval_ = std::min(max_time_interval, poll_interval);
```

Comment:
I think a customer setting the poll interval to `> 5 minutes` would be quite rare, however we should still respect it and instead use that oversized poll interval as our max interval instead of the default one. As a result, the exponential backoff will do nothing, but I think that's fine.